src/Namotion.Reflection/XmlDocsFormatting.cs

```diff
@@ -0,0 +1,129 @@
+        /// <paramref name="formattingMode"/> to generate additional formatting information.
+        /// </summary>
+        /// <param name="stringBuilder">The current <see cref="StringBuilder"/>.</param>
+        /// <param name="element">The <see cref="XElement"/> to append.</param>
+        /// <param name="formattingMode">The <see cref="XmlDocsFormattingMode"/> for generating additional formatting tags.</param>
+        /// <returns>The passed in <paramref name="stringBuilder"/>.</returns>
+        public static StringBuilder AppendFormattedElement(this StringBuilder stringBuilder, XElement element, XmlDocsFormattingMode formattingMode)
+        {
+            // call apropriate formatting function
+            return formattingFunctions[formattingMode](stringBuilder, element);
+        }
+
+        #region No formatting
+        /// <summary>
+        /// Appends the value of <paramref name="element"/> without any formatting information.
+        /// </summary>
+        /// <param name="stringBuilder">The current <see cref="StringBuilder"/>.</param>
+        /// <param name="element">The <see cref="XElement"/> to append.</param>
+        /// <returns>The value of <paramref name="stringBuilder"/>.</returns>
+        private static StringBuilder AppendUnformattedElement(this StringBuilder stringBuilder, XElement element)
+        {
+            stringBuilder.Append(element.Value);
+            return stringBuilder;
+        }
+        #endregion
+        #region HTML formatting
+        private static readonly IDictionary<string, Func<StringBuilder, XElement, StringBuilder>> htmlTagMap =
+            new Dictionary<string, Func<StringBuilder, XElement, StringBuilder>>()
+            {
+                { "c", (sb, e) => AppendSimpleTaggedElement(sb, e, "<pre>", "</pre>") },
+                { "b", (sb, e) => AppendSimpleTaggedElement(sb, e, "<strong>", "</strong>") },
+                { "strong", (sb, e) => AppendSimpleTaggedElement(sb, e, "<strong>", "</strong>") },
+                { "i", (sb, e) => AppendSimpleTaggedElement(sb, e, "<i>", "</i>") }
+            };
+
+        /// <summary>
+        /// Appends the value of <paramref name="element"/> surrounded by the neccessary HTML-tags to maintain its formatting information (if supported).
+        /// </summary>
+        /// <param name="stringBuilder">The current <see cref="StringBuilder"/>.</param>
+        /// <param name="element">The <see cref="XElement"/> to append.</param>
+        /// <returns>The value of <paramref name="stringBuilder"/>.</returns>
+        private static StringBuilder AppendHtmlFormattedElement(StringBuilder stringBuilder, XElement element)
+        {
+            return AppendMapFormattedElement(stringBuilder, element, htmlTagMap);
+        }
+        #endregion
+
+        #region Markdown formatting
+        private static readonly IDictionary<string, Func<StringBuilder, XElement, StringBuilder>> markdownTagMap =
```

Comment:
again `Dictionary<string, Func<StringBuilder, XElement, StringBuilder>>` would be better

Reply:
Changed to `Dictionary<...>`